File: app/src/main/java/com/example/shopkeep/MainActivity.java
package com.example.shopkeep;

import android.content.Intent;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.ImageView;
import android.widget.Toast;

import com.android.volley.AuthFailureError;
import com.android.volley.Request;
import com.android.volley.RequestQueue;
import com.android.volley.Response;
import com.android.volley.VolleyError;
import com.android.volley.toolbox.StringRequest;
import com.android.volley.toolbox.Volley;


import org.json.JSONArray;
import org.json.JSONObject;

import java.util.HashMap;
import java.util.Map;

public class MainActivity extends AppCompatActivity {
    private ImageView imageView;
    private EditText login;
    private EditText password;
    private Button regestrationButton;
    private Button loginButton;
    private static String URL = "http://192.168.1.13/login.php";
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        ImageView imageView = findViewById(R.id.imageView2);
        loginButton = findViewById(R.id.login);
        regestrationButton = findViewById(R.id.regestration);
        imageView.setImageResource(R.drawable.shop);


        login = findViewById(R.id.editText12);
        password = findViewById(R.id.editText11);



        regestrationButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Toast.makeText(getApplicationContext(),"Rejestracja",Toast.LENGTH_LONG);

                openRegestrationActivity();
            }
        });

        loginButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                loginFunction();


            }
        });


    }


    private void openRegestrationActivity(){
        Intent intent = new Intent(getApplicationContext(),regestrationClass.class);
        startActivity(intent);

    }

    private void loginFunction(){

        String pass = password.getText().toString();
        String log = login.getText().toString();
        if(!pass.isEmpty() || !log.isEmpty()){
            loginButton.setVisibility(View.GONE);
            login(log,pass);
        }
        else{
                password.setError("Brak hasła!");
                login.setError("Brak loginu!");

        }

    }

    private void login(final String login,final String password){

        StringRequest stringRequest = new StringRequest(Request.Method.POST,
                URL, new Response.Listener<String>() {
            @Override
            public void onResponse(String response) {
                try {
                    JSONObject jsonObject = new JSONObject(response);
                    String success = jsonObject.getString("success");
                    JSONArray jsonArray = jsonObject.getJSONArray("logins");
                    if(success.equals("1"))
                    {
                       for(int i=0;i<jsonArray.length();i++){
                           JSONObject object = jsonArray.getJSONObject(i);

                           String name = object.getString("login").trim();
                           Toast.makeText(getApplicationContext(),name.toString(),Toast.LENGTH_SHORT);
                           //Przekazanie loginu dalej czyli zalogowanny przyda się podczas dodawania produktu do koszyka
                           //Ale niech narazie tak będzie ;)
                           System.out.println(name);
                           Intent intent = new Intent(getApplicationContext(),menuActivity.class);
                           startActivity(intent);

                       }
                    }
                }
                catch(Exception e){
                    Toast.makeText(getApplicationContext(),"Blad!" + e.toString(),Toast.LENGTH_SHORT);
                    regestrationButton.setVisibility(View.VISIBLE);
                    System.out.println(e.toString());

                }
            }
        }
                , new Response.ErrorListener() {
            @Override
            public void onErrorResponse(VolleyError error) {
                System.out.println(error.toString());
                Toast.makeText(getApplicationContext(),"Blad!" + error.toString(),Toast.LENGTH_SHORT);
                regestrationButton.setVisibility(View.VISIBLE);

            }
        })
        {
            @Override
            protected Map<String, String> getParams() throws AuthFailureError {
                Map<String, String> params = new HashMap<>();
                params.put("login", login);
                params.put("password",password);
                return params;
            }
        };

        RequestQueue requestQueue = Volley.newRequestQueue(getApplicationContext());
        requestQueue.add(stringRequest);

    }
}

File: app/src/main/java/com/example/shopkeep/MainActivity.kt
package com.example.shopkeep


import android.content.Context
import android.content.Intent
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.util.Log
import android.widget.Toast
import com.android.volley.Response
import com.android.volley.toolbox.StringRequest
import com.android.volley.toolbox.Volley
import com.example.shopkeep.Static.Shared
import com.google.android.gms.flags.impl.SharedPreferencesFactory
import kotlinx.android.synthetic.main.activity_main.*
import org.json.JSONObject
import java.util.*

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)


        buttonRegisterShop.setOnClickListener {
            Toast.makeText(applicationContext, "Rejestracja", Toast.LENGTH_LONG).show()
            openRegistrationActivity()
        }
        TextLogin.setOnClickListener { loginFunction() }

        Toast.makeText(this,Shared.getUserName(applicationContext),Toast.LENGTH_SHORT).show()
        editTextLogin.setText(Shared.getUserName(applicationContext))
        editTextPassword.setText(Shared.getPassword(applicationContext))
    }


    private fun openRegistrationActivity() {
        val intent = Intent(applicationContext, RegistrationClass::class.java)
        startActivity(intent)
    }

    private fun loginFunction() {

        val pass = editTextPassword.text.toString()
        val log = editTextLogin.text.toString()

        if (pass.isNotEmpty() || log.isNotEmpty()) {
            login(log, pass)
        } else {
            editTextPassword.error = "Brak hasła!"
            TextLogin.error = "Brak loginu!"

        }

    }

    private fun login(login: String, password: String) {


        val stringRequest = object : StringRequest(Method.POST, URL, Response.Listener
        { response ->

            val responseTrimed = response.substringBefore("<!-- End //]]>")

            val jsonObject = JSONObject(responseTrimed)
            val success = jsonObject.getString("success")
            val jsonArray = jsonObject.getJSONArray("logins")

            if (success == "1")
            {
                for (i in 0 until jsonArray.length())
                {

                    //wykurwia apkę jak coś źle odbierze xd
                    val jsonObjectRecived = jsonArray.getJSONObject(i)

                    val login = jsonObjectRecived.getString("login").trim()

                    //Przekazanie loginu dalej czyli zalogowanny przyda się podczas dodawania produktu do koszyka
                    //Ale niech narazie tak będzie ;)

                    val intent = Intent(applicationContext, MenuActivity::class.java)
                    intent.putExtra("login", login)

                    Shared.save(login,password,applicationContext)

                    startActivity(intent)

                }
            }
            else
            {
                Toast.makeText(applicationContext, "Zły login lub hasło", Toast.LENGTH_SHORT).show()
            }

        },Response.ErrorListener
        {
            Toast.makeText(applicationContext, "Błąd Połączenia! ${it.message}", Toast.LENGTH_SHORT).show()
            startActivity(intent)
        })
        {
            override fun getParams(): Map<String, String>
            {
                val params = HashMap<String, String>()
                params["login"] = login
                params["password"] = password
                return params
            }
        }

        val requestQueue = Volley.newRequestQueue(applicationContext)
        requestQueue.add(stringRequest)

    }

    companion object {
        private const val URL = "http://kryptoprojekt.prv.pl/login.php"
    }
}
